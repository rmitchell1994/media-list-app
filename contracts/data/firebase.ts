interface Firebase {
  apiKey: "AIzaSyCoxTs8fCDpVYS2EH9PFqPMYckuIgmooIY";
  authDomain: "media-list-app.firebaseapp.com";
  projectId: "media-list-app";
  storageBucket: "media-list-app.appspot.com";
  messagingSenderId: "622664374317";
  appId: "1:622664374317:web:c5711b9a9e6bbb946e00a4";
  measurementId: "G-XJM9V52B2V";
}
