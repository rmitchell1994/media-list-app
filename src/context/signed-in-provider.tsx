import React, {
  createContext,
  FC,
  ReactNode,
  useEffect,
  useState,
} from "react";
import firebase from "firebase";

type Props = {
  children: ReactNode;
};

export const SignedInContext = createContext<firebase.UserInfo | null>(null);

const SignedInProvider: FC<Props> = ({ children }: Props) => {
  const [user, setUser] = useState<firebase.UserInfo | null>(null);

  useEffect(() => {
    const userInformation = firebase.auth().currentUser;
    setUser(userInformation);
  });

  return (
    <SignedInContext.Provider value={user}>{children}</SignedInContext.Provider>
  );
};

export default SignedInProvider;
