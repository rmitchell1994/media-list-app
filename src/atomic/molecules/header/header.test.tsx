import React, { FC, useContext } from "react";
import firebase from "firebase";
import Header from "./header";
import { render, screen } from "../../../../test/test-utils";
import { SignedInContext } from "../../../context/signed-in-provider";

const RenderHeader: FC<{ user: firebase.UserInfo | null }> = ({ user }) => (
  <SignedInContext.Provider value={user}>
    <Header />
  </SignedInContext.Provider>
);

describe("Header Component", () => {
  it("should render default component with logout link", () => {
    const { asFragment } = render(
      <RenderHeader
        user={{
          displayName: "name",
          email: "email",
          phoneNumber: null,
          photoURL: null,
          providerId: "provider",
          uid: "uid",
        }}
      />
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText("Log Out")).toBeTruthy();
  });

  it("should render login link", () => {
    render(<RenderHeader user={null} />);

    expect(screen.getByText("Login")).toBeTruthy();
  });
});
