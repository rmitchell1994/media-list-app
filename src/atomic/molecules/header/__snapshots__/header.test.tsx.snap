// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render default component with logout link 1`] = `
<DocumentFragment>
  <header
    class="sc-bdfBwQ jelvnu"
  >
    <div
      class="sc-hKgILt dhzLxW"
    />
    <h1
      class="sc-gsTCUz gvRwRM"
    >
      Media App Header
    </h1>
    <a
      class="sc-dlfnbm ckTqcN"
      href="/account"
    >
      Log Out
    </a>
  </header>
</DocumentFragment>
`;
